### PR TITLE
Convert business units to DB table

### LINKED ISSUE #965: Convert business units to DB table
Currently the business units are coded as choices in employees/models.py. We need to change to (and migrate the data to) a DB table. We'd like to support sub-disciplines and easier changes.

### PR DESCRIPTION
**Pull request by tbaxter-18f**
## Description

Resolves #965 and converts business units to a DB table instead of choices, along with updated tests and fixtures. Also cleans up some tech debt around how orgs are defined and used.

### CODE REVIEW

File: tock/api/views.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ class UserDataSerializer(serializers.Serializer):
     organization = serializers.StringRelatedField()
 
     def get_unit(self,obj):
```
Comment (Jkrzy) [inline tock/api/views.py]: I think there's an opportunity here to drop this altogether and let the `StringRelatedField` handle the work like we're doing for `organization`

Comment (tbaxter-18f) [reply tock/api/views.py]: Initially I had some concerns about scope creep, but after searching I don't see where we actually call `get_unit` anywhere, so may as well drop it.

Comment (tbaxter-18f) [reply tock/api/views.py]: Correction... without this one we throw a serializer error that looks weird enough that I don't want to track it down for this story. Let's log a separate issue

File: tock/hours/views.py [hunk 2]
```diff
@@ -94,21 +94,13 @@ class GeneralSnippetsTimecardSerializer(serializers.Serializer):
     )
     hours_spent = serializers.DecimalField(max_digits=5, decimal_places=2)
     notes = serializers.CharField()
-    unit = serializers.SerializerMethodField()
```
Comment (tbaxter-18f) [inline tock/hours/views.py]: We had already defined `unit` on L88

File: tock/employees/admin.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ class UserDataAdmin(admin.ModelAdmin):
     search_fields = ('user__username',)
 
     def unit_info(self, obj):
-        return obj.get_unit_display()
+        return obj.unit.name
```
Comment (amymok) [inline tock/employees/admin.py]: This needs to check for `None` object.  It is currently breaking the admin interface.

Comment (tbaxter-18f) [reply tock/employees/admin.py]: Nice catch Amy! Updated, and added a comment.

**Comment by amymok** [comment]
Merging this PR, something is not quite working in Snyk.  There's no manifest changes.

**Pull request merged**